pass variables overrides via environment variables (#16246)
DataDog/documentation: content/en/synthetics/private_locations/configuration.md

`maxNbRedirects`
: **Type**: Number <br>
**Default**: `20`<br>
Maximum number of redirections for HTTP and WebSocket tests.

`variableOverride`
: **Type**: String <br>

> Overrides the variables used in tests running on the Private Location. Format: `SECRET_VAR=value`.

Author: ```suggestion
Overrides the variables used in tests running on the Private Location. Format: `LOCAL_VAR=value`.
```

> All variables imported this way will be obfuscated.

Reviewer: ```suggestion
All variables imported this way are obfuscated.
```

`environmentVariableOverride`
: **Type** String <br>

> Allows the usage of environment variables in the Private Location. It requires the environment variables to be imported in the containerized environment.

Author: ```suggestion
Overrides variables used in tests running on the Private Location with environment variables. It requires the environment variables to be imported in the containerized environment.
```

> Example for Docker: `docker run --env SECRET=value --environmentVariableOverride synthetics-private-location-worker`.

Reviewer 2: ```suggestion
Example for Docker: `docker run --env SECRET=value datadog/synthetics-private-location-worker --environmentVariableOverride`.
```

Author: btw should it be --env SECRET=value or --env SECRET ?

Reviewer 2: Both work:

- `--env SECRET` will expose `SECRET=<value from the host>` in the container
- `--env SECRET=value` will expose `SECRET=value` in the container

Author: so option 2 = using `variableOverride`

Reviewer 2: Yes

> Example for Docker: `docker run --env SECRET=value gcr.io/datadoghq/synthetics-private-location-worker --environmentVariableOverride SECRET`.

Author: ```suggestion
Example for Docker: `docker run --env VARIABLE gcr.io/datadoghq/synthetics-private-location-worker --environmentVariableOverride SECRET`.
```

> All variables imported this way will be obfuscated. 

Reviewer: ```suggestion
All variables imported this way are obfuscated. 
```
